frontend/src/metabase-lib/lib/Dimension.js

```diff
@@ -858,9 +858,19 @@ export class AggregationDimension extends Dimension {
   }
 
   field() {
-    // FIXME: it isn't really correct to return the unaggregated field. return a fake Field object?
-    const dimension = this.aggregation().dimension();
-    return dimension ? dimension.field() : super.field();
+    const aggregation = this.aggregation();
+    if (aggregation) {
+      const dimension = aggregation.dimension();
+      const field = dimension && dimension.field();
+      return new Field({
+        display_name: aggregation.displayName(),
```

Comment:
I think there's some overlap between this and `AggregationDimension`'s `column` method. Maybe we should extract that somehow.

e.x. `column` uses `UNAGGREGATED_SPECIAL_TYPES` to avoid carrying over `special_type` in some cases.

Reply:
Yeah good call. `super.column` calls `field` so I can move that logic here to simplify `this.column`.